support-workers/src/main/scala/com/gu/zuora/ZuoraSubscriptionCreator.scala

```diff
@@ -0,0 +1,65 @@
+package com.gu.zuora
+
+import com.gu.WithLoggingSugar._
+import com.gu.monitoring.SafeLogger
+import com.gu.support.workers._
+import com.gu.support.zuora.api.response.{ZuoraAccountNumber, ZuoraSubscriptionNumber}
+import com.gu.support.zuora.api.{SubscribeItem, _}
+import com.gu.support.zuora.domain.DomainSubscription
+import com.gu.zuora.ZuoraSubscriptionCreator.checkSingleResponse
+import org.joda.time.DateTime
+
+import java.util.UUID
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ZuoraSubscriptionCreator(
+  zuoraService: ZuoraSubscribeService,
+  now: () => DateTime,
+  userId: String,
+  requestId: UUID,
+) {
+
+  def preview(
+    subscribeItem: SubscribeItem,
+    billingPeriod: BillingPeriod,
+  ): Future[PaymentSchedule] =
+    PreviewPaymentSchedule.preview(subscribeItem, billingPeriod, zuoraService, checkSingleResponse)
```

Comment:
we could always inline `PreviewPaymentSchedule.preview` one day as this is the only place it's used

Reply:
same goes for the below `ensureSubscriptionCreated` function I'm sure it could get simplified some day